docs/sql-ref-syntax-qry-window.md

```diff
@@ -19,4 +19,188 @@ license: |
+  | Lisa|      Sales| 10000|   35|
+  | Alex|      Sales| 30000|   33|
+  +-----+-----------+------+-----+
+
+SELECT name, dept, RANK() OVER (PARTITION BY dept ORDER BY salary) AS rank FROM employees;
+  +-----+-----------+------+----+
+  | name|       dept|salary|rank|
+  +-----+-----------+------+----+
+  | Lisa|      Sales| 10000|   1|
+  | Alex|      Sales| 30000|   2|
+  | Evan|      Sales| 32000|   3|
+  | Fred|Engineering| 21000|   1|
+  |  Tom|Engineering| 23000|   2|
+  |Chloe|Engineering| 23000|   2|
+  | Paul|Engineering| 29000|   4|
+  |Helen|  Marketing| 29000|   1|
+  | Jane|  Marketing| 29000|   1|
+  | Jeff|  Marketing| 35000|   3|
+  +-----+-----------+------+----+
+
+SELECT name, dept, DENSE_RANK() OVER (PARTITION BY dept ORDER BY salary ROWS BETWEEN
+    UNBOUNDED PRECEDING AND CURRENT ROW) AS dense_rank FROM employees;
+  +-----+-----------+------+----------+
+  | name|       dept|salary|dense_rank|
+  +-----+-----------+------+----------+
+  | Lisa|      Sales| 10000|         1|
+  | Alex|      Sales| 30000|         2|
+  | Evan|      Sales| 32000|         3|
+  | Fred|Engineering| 21000|         1|
+  |  Tom|Engineering| 23000|         2|
+  |Chloe|Engineering| 23000|         2|
+  | Paul|Engineering| 29000|         3|
+  |Helen|  Marketing| 29000|         1|
+  | Jane|  Marketing| 29000|         1|
+  | Jeff|  Marketing| 35000|         2|
+  +-----+-----------+------+----------+
+
+SELECT name, dept, age, CUME_DIST() OVER (PARTITION BY dept ORDER BY age
+    RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) as cume_dist FROM employees;
+  +-----+-----------+------+------------------+
+  | name|       dept|age   |         cume_dist|
+  +-----+-----------+------+------------------+
+  | Alex|      Sales|    33|0.3333333333333333|
+  | Lisa|      Sales|    35|0.6666666666666666|
+  | Evan|      Sales|    38|               1.0|
+  | Paul|Engineering|    23|              0.25|
+  |Chloe|Engineering|    25|              0.75|
+  | Fred|Engineering|    28|              0.25|
+  |  Tom|Engineering|    33|               1.0|
+  | Jane|  Marketing|    28|0.3333333333333333|
+  | Jeff|  Marketing|    38|0.6666666666666666|
+  |Helen|  Marketing|    40|               1.0|
+  +-----+-----------+------+------------------+
+
+SELECT name, dept, salary, MIN(salary) OVER (PARTITION BY dept ORDER BY salary) AS min
+    FROM employees;
+  +-----+-----------+------+-----+
+  | name|       dept|salary|  min|
+  +-----+-----------+------+-----+
+  | Lisa|      Sales| 10000|10000|
+  | Alex|      Sales| 30000|10000|
+  | Evan|      Sales| 32000|10000|
+  |Helen|  Marketing| 29000|29000|
+  | Jane|  Marketing| 29000|29000|
+  | Jeff|  Marketing| 35000|29000|
+  | Fred|Engineering| 21000|21000|
+  |  Tom|Engineering| 23000|21000|
+  |Chloe|Engineering| 23000|21000|
+  | Paul|Engineering| 29000|21000|
+  +-----+-----------+------+-----+
+
+SELECT name, salary,
+    LAG(salary) OVER (PARTITION BY dept ORDER BY salary) as lag,
```

Comment:
as -> AS 
but definitely don't change it just for that. Looks fine. I'll merge shortly